packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -57,6 +57,8 @@ function getTopLevelCallbackBookKeeping(
   targetInst: Fiber | null,
   ancestors: Array<Fiber>,
 } {
+  const topLevelType = unsafeCastStringToDOMTopLevelType(nativeEvent.type);
```

Comment:
What's the guideline for when it's "safe" to call it? How do we know it's okay here?

Reply:
I think it's safe here because we know the nativeEvent type is always a DOMTopLevelType. 

This is otherwise only used to cast the top level types themselves when they are created, like:

https://github.com/facebook/react/blob/f6fb03edffcc5918ad2f37ed45bdc7c0c3b3d199/packages/react-dom/src/events/DOMTopLevelEventTypes.js#L25-L31

I wonder if there is an existing concept of a DOM event's type string. We don't really have top level types in anymore: just the browser standard event names.

Reply:
Should have a comment explaining why then.

Reply:
Commented! Part of me wonders if these could be annotated using the actual event type declarations here:

https://github.com/facebook/flow/blob/08f854876cc72165f1f7a2e023bd8e79e8a7fa02/lib/dom.js#L144-L153

If that were the case, we could remove the concept of a top level type and just make it an DOM event type